### PR TITLE
Stream Updates

### PR DESCRIPTION
**Pull request by MattEqualsCoder**
- Increased fequency of polling viewers for content updates from 25% of the time to 33% of the time
- Added logic config for the left sand pit requiring the spring ball and hi jump boots
- Tracking bosses as quick killed or without taking damage will add content

### CODE REVIEW

**Comment by MattEqualsCoder** [comment]
Currently Skelebro thinks that the spring ball requires high jump boots to work properly for this, so I won't merge this until that's confirmed or denied.

**Comment by Vivelin** [comment]
Based on the [wall jump logic document](https://docs.google.com/document/d/12zxeKyZT1LOMpnxzbA2dIVYEiZSFe3iXbPZrazd6BQs/edit) Pink wrote, Hi Jump is required to circumvent wall jumps, with spring ball making it easier.

File: src/Randomizer.SMZ3.Tracking/VoiceCommands/BossTrackingModule.cs [hunk 1]
```diff
@@ -106,5 +143,22 @@ private GrammarBuilder GetMarkBossAsNotDefeatedRule()
...
         }
+
+        private GrammarBuilder GetBossDefeatedWithContentRule()
+        {
+            var bossNames = GetBossNames();
+            var beatBoss = new GrammarBuilder()
+                .Append("Hey tracker,")
+                .OneOf("I beat", "I defeated", "I beat off", "I killed")
+                .Append(BossKey, bossNames)
+                .OneOf("Without getting hit", "Without taking damage", "And didn't get hit", "And didn't take damage");
```
Comment (Vivelin) [inline src/Randomizer.SMZ3.Tracking/VoiceCommands/BossTrackingModule.cs:154]: + "in one cycle" for consistency?

File: src/Randomizer.SMZ3.Tracking/tracker.json [hunk 2]
```diff
@@ -1428,6 +1428,10 @@
     "DungeonBossNotYetCleared": [
       "But you never defeated {1} in the first place."
     ],
+    "DungeonBossClearedAddContent": [
+      "That was some impressive work.",
+      "Good job. I didn't think you had it in you."
```
Comment (Vivelin) [inline src/Randomizer.SMZ3.Tracking/tracker.json:1433]: 😂

**Review by Vivelin — APPROVED** [review]

**Comment by MattEqualsCoder** [comment]
> Based on the [wall jump logic document](https://docs.google.com/document/d/12zxeKyZT1LOMpnxzbA2dIVYEiZSFe3iXbPZrazd6BQs/edit) Pink wrote, Hi Jump is required to circumvent wall jumps, with spring ball making it easier.
> 
> I do have an unmerged PR with most of these changes sitting around. Maybe it's worth merging that instead of these logic changes? It'll need a second look, but I think I included this one.
Well, this was specifically called out by betus on the stream as something he'd be interested in, so I figured it would be good to add standalone. That being said, I forgot about the wall jump changes. I'll try to start looking into that so we can get it merged in.

I did some tests, and it in fact does appear that hi jump boots are required for this, so I updated the logic accordingly.

**Comment by Vivelin** [comment]
Fair enough. I haven't caught up yet.

**Review by Vivelin — DISMISSED** [review]

**Comment by Vivelin** [comment]
There's a fun opportunity for sass by keeping track of the last time Kraid was defeated with the quick kill. For example, if you track the quick kill a second time, you could have tracker tell you _exactly_ how long it's been since the last one, or remind you how long it's been if you didn't get it.

I'd love to add more highly-specific flavor/sass like this, but all the configurability makes it a little harder than it needs to be. Something to think about, I suppose.

**Review by Vivelin — APPROVED** [review]

**Comment by MattEqualsCoder** [comment]
@Vivelin Yeah, agreed. I think some more advanced boss configs could be useful for various things. We'd also need to think of the best way to keep track of those stats between sessions.

**Pull request merged**